packages/docusaurus/src/client/exports/BrowserOnly.tsx

```diff
@@ -15,9 +15,7 @@ function BrowserOnly({
   children?: () => JSX.Element;
   fallback?: JSX.Element;
 }): JSX.Element | null {
-  const {isClient} = useDocusaurusContext();
-
-  if (isClient && children != null) {
+  if (ExecutionEnvironment.canUseDOM && children != null) {
```

Comment:
I don't think this is the solution, this basically reverts your changes in the previous PR and the hydration problem will be back.

Previously we had:

```js
function BrowserOnly({
  children,
  fallback
}: {
  children?: () => JSX.Element,
  fallback?: JSX.Element
}): JSX.Element | null {
  if (!ExecutionEnvironment.canUseDOM || children == null) {
    return fallback || null;
  }

  return <>{children()}</>;
}
```

It looks the same as this PR, except the condition is inverted

Reply:
Ah, yes, you're right, I mistakenly tested with `fallback`, in which case we don't have hydration issue with any implementation.